tests/test_add_id.py

```diff
@@ -18,26 +18,37 @@
 
 import nemo_curator as nc
 from nemo_curator.datasets import DocumentDataset
+from nemo_curator.utils.import_utils import gpu_only_import, is_unavailable
 
+cudf = gpu_only_import("cudf")
+is_cudf_available = not is_unavailable(cudf)
 
-def list_to_dataset(documents, col_name="text", npartitions=2):
+
+def list_to_dataset(documents, col_name="text", npartitions=2, backend="pandas"):
     data = {col_name: documents}
     pdf = pd.DataFrame(data)
-
-    return DocumentDataset(dd.from_pandas(pdf, npartitions=npartitions))
+    ddf = dd.from_pandas(pdf, npartitions=npartitions)
+    if backend == "cudf" and is_unavailable(cudf):
+        pytest.skip("cuDF is not installed or importable.")
```

Comment:
Might be better to raise an error here instead of skipping.

Reply:
Agreed: https://github.com/NVIDIA/NeMo-Curator/pull/479/commits/dc3a13836d9fbfc72d4ec614df9dc19b192c2ff9